provisioning/apache2-vhost.conf

```diff
@@ -0,0 +1,83 @@
+<VirtualHost *:80>
+    ServerAdmin webmaster@localhost
+    ServerName prod.dissem.in
+    ServerAlias dissem.in
+
+    Redirect permanent / https://dissem.in/
+</VirtualHost>
+
+<VirtualHost *:443>
+    ServerAdmin webmaster@localhost
+    ServerName dissem.in
+    ServerAlias prod.dissem.in
+
+    # SSL configuration
+    # =================
+    SSLEngine on
+    SSLCertificateFile    /etc/letsencrypt/live/dissem.in/cert.pem
+    SSLCertificateKeyFile /etc/letsencrypt/live/dissem.in/privkey.pem
+    SSLCertificateChainFile /etc/letsencrypt/live/dissem.in/fullchain.pem
+    # Use HSTS
+    # TODO: Include subdomains? Preload?
+    Header always set Strict-Transport-Security "max-age=15768000"
```

Comment:
This header states that `dissem.in` is available over HTTPS and tells the browser to remember this for 6 months. Therefore, next time the user is sent to the HTTP version (through a link for instance), the browser will automatically upgrade to HTTPS transparently, before making any HTTP call.

The downside is that we should always have a working SSL stack (or the site will not be browsable at all, one cannot overload HSTS policy errors). This should be fine with the current Let's Encrypt setup and a bit of monitoring.

Question here:
- [ ] Do we want to include subdomains? So that upon visiting `dissem.in`, any `*.dissem.in` will be accessed on HTTPS next time. I think so, but we should make sure we have HTTPS available on all subdomains. `dev.dissem.in` and `association.dissem.in` are both fine. Do we have any others?
- [ ] Do we want to include preloading? So that whenever a user visits our website, the rule is sent to a central list and all users will now make queries in HTTPS by default? I think so as well :)

Reply:
On my own web properties I've always been wary of enabling HSTS, because of the significant downside (broken SSL setup means that the website is completely unreachable for all users -- HSTS also mandates that browsers must not allow users to override errors). Broken HTTPS (e.g., on subdomains) has happened in the past, so that's a genuine risk. My understanding of HSTS is also that this choice is "irreversible" in the sense that if we change our mind about this, we will be stuck with clients expecting HTTPS to work until expiration of the max-age value.

The upside is that user traffic is encrypted, but it's not like I see any real benefit to do this for Dissemin traffic.

I don't know, though: as Let's Encrypt has been around for a long time now and HTTPS is mostly working fine, maybe I'm just being too risk-averse here.

Reply:
That's precisely the main issue with having HSTS (it being irreversible etc). For the record, the current SSL configuration is based on Mozilla's recommendation (https://mozilla.github.io/server-side-tls/ssl-config-generator/) and the oldest browsers supported by this configuration probably already see a broken Dissemin interface due to lacking some modern CSS / JS abilities.

My opinion is that for a website such as Dissemin aiming at a broad audience, HTTPS and HSTS should be implemented. Then, this implies we should have certificates monitoring and a proper SSL setup (but we should anyways have such monitoring in my opinion if we target broad audience :).

Reply:
Checking with the recent DNS entries update, I found doai and the server hostnames which do not have a valid SSL cert. The rest should be fine.

My proposal would be to keep

```
Header always set Strict-Transport-Security "max-age=15768000; preload"
```

This means that upon visiting `dissem.in`, the info that HTTPS is available and should be used will be kept for 15768000 seconds (6 months) and this info will be sent to the HSTS preload list (built into most recent browsers).

I would not add `includeSubDomains` as we might miss HTTPS on subdomains or not monitor them correctly.

What do you think?